src/backend/marsha/account/tests/test_views.py

```diff
@@ -0,0 +1,236 @@
+
+
+class AccountViewTestCase(BaseAuthenticationTestCase):
+    """Test case for the full login/logout process."""
+
+    def test_login_get(self):
+        """Assert the login view is reachable when not logged in."""
+        response = self.client.get(
+            reverse("account:login"),
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, "Username")
+        self.assertContains(response, "Password")
+        self.assertContains(response, "Forgotten your password or username?")
+
+    def test_login_post_fails(self):
+        """Assert the login view displays an error for non-matching username/password."""
+        response = self.client.post(
+            reverse("account:login"),
+            {"username": self.existing_user.username, "password": "wrong_password"},
+        )
+
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, "Username")
+        self.assertContains(response, "Password")
+        self.assertContains(response, "Forgotten your password or username?")
+        self.assertContains(
+            response,
+            "Please enter a correct username and password. "
+            "Note that both fields may be case-sensitive.",
+        )
+        self.assertIsNotAuthenticated()
+
+    def test_login_post_success(self):
+        """Assert the login view redirects properly to the main page."""
+        response = self.client.post(
+            reverse("account:login"),
+            {"username": self.existing_user.username, "password": "password"},
+        )
+
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, "/")
+        self.assertIsAuthenticated()
+
+    def test_logout_post_success(self):
+        """Assert the logout view redirects properly to the main page."""
+        self._login_user()
+
+        response = self.client.post(reverse("account:logout"))
+
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, "/")
+        self.assertIsNotAuthenticated()
+
+    def test_logout_get_success(self):
+        """Assert the logout view redirects properly to the main page."""
+        self._login_user()
+
+        response = self.client.get(reverse("account:logout"))
+
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, "/")
+        self.assertIsNotAuthenticated()
+
+    def test_logout_get_success_if_not_logged_in(self):
+        """Assert the logout view works even when the user is not logged in."""
+        self.assertIsNotAuthenticated()
+        response = self.client.get(reverse("account:logout"))
+
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, "/")
+        self.assertIsNotAuthenticated()
+
+    def test_password_reset_get(self):
+        """Assert the password reset view is available."""
+        response = self.client.get(reverse("account:password_reset"))
```

Comment:
Should you test if the user is authenticated or not before doing the request ? Does this view should be reachable if the user is already authenticated ?

Reply:
Using links, you can't really reach this view when authenticated but I'm not sure why we would prevent someone, already logged in, to access this and ask for a password reset. Do you think it could prevent mistakes ?